src/unit_tests/test_fim_db.c

```diff
@@ -973,8 +1110,8 @@ void test_fim_db_get_paths_from_inode_multiple_unamatched_rows(void **state) {
         snprintf(buffer, 10, "Path %d", i + 1);
         will_return(__wrap_sqlite3_column_text, strdup(buffer));
     }
-    will_return(__wrap_sqlite3_step, SQLITE_ERROR);
-    expect_string(__wrap__merror, formatted_msg, "Error in fim_db_get_paths_from_inode(): Unmatched number of rows in queries");
+    will_return(__wrap_sqlite3_step, SQLITE_ROW);
+    expect_string(__wrap__minfo, formatted_msg, "The count returned is smaller than the actual elements. This shouldn't happen.");
     wraps_fim_db_check_transaction();
     char **paths;
     paths = fim_db_get_paths_from_inode(test_data->fim_sql, 1, 1);
```

Comment:
Tests on fim_db_get_paths_from_inode are leaking the returned paths, I recommend writing a specific teardown for these tests in order to free them.

Reply:
Fixed